sender/rabbitmq.go

```diff
@@ -0,0 +1,136 @@
+/*
+ * skogul, rabbitmq producer/sender
+ *
+ * Copyright (c) 2023 Telenor Norge AS
+ * Author(s):
+ *  - Kamil Oracz <kamil.oracz@telenor.no>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301  USA
+ */
+
+package sender
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	amqp "github.com/rabbitmq/amqp091-go"
+	"github.com/telenornms/skogul"
+	"github.com/telenornms/skogul/encoder"
+)
+
+type Rabbitmq struct {
+	Username string            `doc:"Username for rabbitmq instance"`
+	Password string            `doc:"Password for rabbitmq instance"`
+	Host     string            `doc:"Hostname for rabbitmq instance. Fallback is localhost"`
+	Port     string            `doc:"Port for rabbitmq instance. Fallback is 5672"`
+	Queue    string            `doc:"Queue to write to"`
+	Encoder  skogul.EncoderRef `doc:"Encoder to use. Fallback is json"`
+	Timeout  int               `doc:"Timeout for rabbitmq instance connection. Fallback is 10 seconds."`
+	channel  *amqp.Channel
+	once     sync.Once
+}
+
+func (r *Rabbitmq) init() {
+	if r.Username == "" || r.Password == "" {
+		fmt.Errorf("Error missing username or password")
+		return
+	}
```

Comment:
I would prefer to have this in a `Verify` function instead of in the `init` function. Verify is called when Skogul starts, and will therefore cause Skogul to exit with an error message immediately if this is misconfigured. If it is checked first in `init`, this will not trigger an error before the first message is attempted sent (which is when init() runs).